pages/community/community-contribute-to-precice.md

```diff
@@ -71,23 +68,26 @@ First time working with Git? Watch a [short lecture on the GitHub workflow](http
 Our tutorials generally follow a file structure similar to this:
 
 ```bash
-- <tutorial>/               # e.g. perpendicular-flap/
-  - README.md               # description of the case
-  - precice.config.xml      # a works-with-all preCICE configuration file
-  - clean-tutorial.sh       # a symbolic link (see ../tools/)
-  - <visualization scripts> # gnuplot or simple Python scripts
-  - images/                 # any images used by the documentation
-  - <participant1-solver1>/ # e.g. fluid-openfoam/
-    - run.sh                # a short script to run the solver1 case
-    - clean.sh              # a short script to clean the solver1 case
+- <tutorial>/                     # e.g. perpendicular-flap/
+  - README.md                     # description of the case
+  - precice.config.xml            # a works-with-all preCICE configuration file
+  - clean-tutorial.sh             # a symbolic link (see ../tools/)
+  - <visualization scripts>       # gnuplot or simple Python scripts
+  - images/                       # any images used by the documentation
+  - solver-<code>                 # any configurable, tutorial-specific code, e.g., solver-fenics
+  - <participant1-solver1>/       # e.g. fluid-openfoam/
+    - run.sh                      # a short script to run the solver1 case
+    - clean.sh                    # a short script to clean the solver1 case
     - <the solver1 files>
-  - <participant2-solver2>/ # e.g. solid-dealii/
+  - <participant2-solver2>/       # e.g. solid-dealii/
     - run.sh
     - clean.sh
     - <the solver2 files>
```

Comment:
I think it would be generally desirable that solvers write their output into a dedicated `output` folder. Maybe this is hard to realize, but the nutils-output-situation is in my opinion quite terrible.

Reply:
We do not have much control over this. Not having the results in the same directory is a good practice that we can always aim for, but every solver has its own system (think OpenFOAM and CalculiX). I don't think we need to specify here what each solver does internally.

Reply:
I added a "it would be helpful if" in the (new) section about tutorial-specific codes. This is anyway what you meant, and not a general rule for every solver out there.